File: Flutter/app_eventos/lib/core/models/dto/register_dto.dart
class RegisterDto {
  final String name;
  final String lastName;
  final int? phone;
  final String? birthDate; // yyyy-MM-dd
  final String? city;
  final String? country;
  final String username;
  final String email;
  final String password;
  final String? role;

  RegisterDto({
    required this.name,
    required this.lastName,
    required this.phone,
    required this.birthDate,
    required this.city,
    required this.country,
    required this.username,
    required this.email,
    required this.password,
    this.role,
  });

  Map<String, dynamic> toJson() => {
        'name': name,
        'lastName': lastName,
        'phone': phone,
        'birthDate': birthDate,
        'city': city,
        'country': country,
        'username': username,
        'email': email,
        'password': password,
        'role': role,
      };

  factory RegisterDto.fromJson(Map<String, dynamic> json) {
    return RegisterDto(
      name: json['name'],
      lastName: json['lastName'],
      phone: json['phone'] != null ? int.tryParse(json['phone'].toString()) : null,
      birthDate: json['birthDate'],
      city: json['city'],
      country: json['country'],
      username: json['username'],
      email: json['email'],
      password: json['password'],
      role: json['role'],
    );
  }
}

File: TFC-AppEventos/TFC.AppEventos.Application.DTO/RegisterDTO.cs
using System.ComponentModel.DataAnnotations;

namespace TFC.AppEventos.Application.DTO
{
    public class RegisterDTO
    {

        public string Name { get; set; }
        public string LastName { get; set; }
        public int? Phone { get; set; }
        public DateOnly? BirthDate { get; set; }
        public string? City { get; set; }
        public string? Country { get; set; }
        public string Username { get; set; }
        [EmailAddress]
        public string Email { get; set; }
        public string Password { get; set; }
        public string? Role { get; set; }
    }
}
